activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -243,7 +243,9 @@ def format_options(options)
       end
 
       def remove_prefix_and_suffix(table)
-        table.gsub(/^(#{@options[:table_name_prefix]})(.+)(#{@options[:table_name_suffix]})$/,  "\\2")
+        prefix = Regexp.escape(@options[:table_name_prefix].to_s)
+        suffix = Regexp.escape(@options[:table_name_suffix].to_s)
+        table.sub(/\A(?:#{prefix})(.+)(?:#{suffix})\z/, "\\1")
```

Comment:
Thanks for the review. 

This regexp was suggested by @kamipo at https://github.com/rails/rails/issues/30044#issuecomment-319745737 then I do not have a strong opinion or technical background for having two separate non-capturing groups around the prefix/suffix. 

@kamipo Would you give us some comments for this review? Thanks.